kernel/dev/tty/tty.cpp

```diff
@@ -96,31 +98,40 @@ static size_t ansi_values_index = 0;
 #define PUSH_VAL(VAL) ansi_values[ansi_values_index++] = (VAL)
 #define POP_VAL() ansi_values[--ansi_values_index]
 
-void putchar(char c) {
+int putchar(char c) {
     switch (ansi_state) {
         case Normal: // print the character out normally unless it's an ESC
             if (c != ESC) break;
             ansi_state = Esc;
-            return;
+            return c;
```

Comment:
This may be a good candidate for `goto end` instead of many `return c`s

Reply:
That's fair. Is end already a label or will I need to update all of the returns to `goto end`?

Reply:
You'll need to add it.